### PR TITLE
botoes voltar e decisao e pause, e teste deles

### PR DESCRIPTION
**Pull request by pietro1704**
componentes botao pause e voltar e inicio testes deles

### CODE REVIEW

File: challengeFinalComponents/PrimaryButton/BackButton.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    public override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setupButton()
+    }
+    
+    public init(title: String, for state: UIControl.State = .normal) {
+        super.init(frame: .zero)
+        setupButton()
+        setTitle(title, for: state)
+        #warning("ou title=Voltar hard coded?")
```
Comment (guilhermeluisdomingues) [inline challengeFinalComponents/PrimaryButton/BackButton.swift:26]: why this warning?

File: challengeFinalComponents/PrimaryButton/BackButton.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        addTarget(self, action: #selector(handleTap), for: .touchUpInside)
+    }
+    
+    @objc private func handleTap() {
+        self.delegate?.buttonPressed(tag)
+    }
+    
+    private func setupButton() {
+        //transparent background
+        self.backgroundColor = .none
```
Comment (guilhermeluisdomingues) [inline challengeFinalComponents/PrimaryButton/BackButton.swift:36]: shouldn't be .clear? 
this comment is useless

File: challengeFinalTests/Components/DecisionTextViewTests.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//  Created by Guilherme Domingues on 14/10/20.
+//
+
+import XCTest
+@testable import challengeFinal
+import SnapshotTesting
+
+class DecisionTextViewTests: XCTestCase {
+    func testStackView () {
+        let textView = RegularTextView()
```
Comment (guilhermeluisdomingues) [inline challengeFinalTests/Components/DecisionTextViewTests.swift:14]: This shouldn't be DecisionTextView() ?

File: challengeFinalComponents/PrimaryButton/PauseButton.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+class PauseButton: UIButton {
+    public weak var delegate: PauseButtonDelegate?
+    
+    public override init(frame: CGRect) {
+        super.init(frame: .zero)
+        setupButton(for: .normal)
+    }
+    
+    public init(for state: UIControl.State = .normal) {
```
Comment (guilhermeluisdomingues) [inline challengeFinalComponents/PrimaryButton/PauseButton.swift:22]: how can we change the color of this button?

File: challengeFinal/Flow/Story/ViewModel/DecisionViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,12 @@
+//
+//  DecisionViewModel.swift
+//  challengeFinal
+//
+//  Created by Pietro Pugliesi on 22/10/20.
+//
+
+import UIKit
+
+public class DecisionViewModel: NSObject {
```
Comment (guilhermeluisdomingues) [inline challengeFinal/Flow/Story/ViewModel/DecisionViewModel.swift:10]: this class isn't being used

**Review by guilhermeluisdomingues — CHANGES_REQUESTED** [review]

File: challengeFinalComponents/PrimaryButton/BackButton.swift (hunk 2 above)
Comment (pietro1704) [reply challengeFinalComponents/PrimaryButton/BackButton.swift:36]: updated

**Comment by pietro1704** [comment]
corrigido na branch de merge

**Pull request closed without merging**